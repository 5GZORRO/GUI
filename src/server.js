/* eslint-disable */
import { createServer, Model, Factory, belongsTo, hasMany, RestSerializer } from 'miragejs'
import faker from 'faker'

export function makeServer({ environment = 'development' } = {}) {
  const server = createServer({
    environment,

    serializers: {
      application: RestSerializer
      /* resource: RestSerializer.extend({
        serializeIds: false,
        include: ['resourceSpecification', 'category', 'resourcePhysicalCapabilities', 'resourceVirtualCapabilities'],
        embed: true
      }),
      resourcePhysicalCapabilitie: RestSerializer.extend({
        serializeIds: false,
        include: ['hardwareCapabilities', 'feature'],
        embed: true
      }),
      resourceVirtualCapabilitie: RestSerializer.extend({
        serializeIds: false,
        include: ['virtualCapabilities'],
        embed: true
      }) */
    },

    models: {
      pagedGovernanceProposal: Model,
      resourceCandidate: Model,

      category: Model,

      resourceSpecification: Model,

      resourcePhysicalCapabilitie: Model.extend({
        hardwareCapabilities: hasMany(),
        feature: belongsTo()
      }),

      hardwareCapabilitie: Model,
      feature: Model,

      resourceVirtualCapabilitie: Model.extend({
        virtualCapabilities: hasMany()
      }),
      virtualCapabilitie: Model
    },

    factories: {
      pagedGovernanceProposal: Factory.extend({
        proposalId() {
          return faker.random.uuid()
        },
        status() {
          return faker.random.arrayElement(['PROPOSED', 'APPROVED', 'REJECTED'])
        },
        actionType() {
          return faker.random.arrayElement([
            'ONBOARD_STAKEHOLDER',
            'NEW_LEGAL_PROSE_TEMPLATE',
            'ARCHIVE_LEGAL_PROSE_TEMPLATE',
            'SLA_DISPUTE'
          ])
        },
        statusUpdated() {
          return faker.date.recent()
        },
        actionParams() {
          return {
            entityIdentityId: faker.random.uuid(),
            evidence: faker.random.word()
          }
        }
      }),

      resourceCandidate: Factory.extend({
        baseType() {
          return 'string'
        },
        schemaLocation() {
          return 'string'
        },
        type() {
          return 'string'
        },
        category: [
          {
            baseType() {
              return 'string'
            },
            schemaLocation() {
              return 'string'
            },
            type() {
              return 'string'
            },
            href() {
              return faker.internet.url()
            },
            id() {
              return faker.random.uuid()
            },
            name() {
              return faker.lorem.sentence()
            },
            version() {
              return faker.system.semver()
            }
          }
        ],
        description() {
          return faker.lorem.sentences()
        },
        href() {
          return faker.internet.url()
        },
        id() {
          return faker.random.uuid()
        },
        lastUpdate() {
          return faker.date.past()
        },
        lifecycleStatus() {
          return faker.random.arrayElement(['pending', 'new', 'archive', 'delete'])
        },
        name() {
          return faker.lorem.sentence()
        },

        resourceSpecification() {
          return {
            referredType() {
              return 'string'
            },
            baseType() {
              return 'string'
            },
            schemaLocation() {
              return 'string'
            },
            type() {
              return 'string'
            },
            href() {
              return faker.internet.url()
            },
            id() {
              return faker.random.uuid()
            },
            name() {
              return faker.commerce.productName()
            },
            version() {
              return faker.system.semver()
            }
          }
        },

        validFor() {
          return {
            endDateTime() {
              return faker.date.past()
            },
            startDateTime() {
              return faker.date.future()
            }
          }
        },

        version() {
          return faker.random.uuid()
        }
      }),

      resource: Factory.extend({
        name() {
          return faker.lorem.word()
        },
        description() {
          return faker.commerce.productDescription()
        },
        version() {
          return faker.system.semver()
        },
        validFor() {
          return faker.date.past()
        },
        ownerdid() {
          return faker.random.uuid()
        }
      }),

      category: Factory.extend({
        name() {
          return faker.lorem.word()
        },
        type() {
          return faker.name.jobType()
        },
        version() {
          return faker.system.semver()
        }
      }),

      resourceSpecification: Factory.extend({
        name() {
          return faker.lorem.word()
        },
        description() {
          return faker.commerce.productDescription()
        },
        version() {
          return faker.system.semver()
        }
      }),

      resourcePhysicalCapabilitie: Factory.extend({
        name() {
          return faker.lorem.word()
        },
        description() {
          return faker.commerce.productDescription()
        }
      }),

      feature: Factory.extend({
        bundled() {
          return faker.random.boolean()
        },
        type() {
          return faker.random.word()
        }
      }),

      hardwareCapabilitie: Factory.extend({
        value() {
          return faker.random.float()
        },
        quota() {
          return faker.random.word()
        }
      }),

      resourceVirtualCapabilitie: Factory.extend({
        name() {
          return faker.lorem.word()
        },
        description() {
          return faker.commerce.productDescription()
        },
        isMaster() {
          return faker.random.boolean()
        },
        type() {
          return faker.database.type()
        }
      }),

      virtualCapabilitie: Factory.extend({
        capValue() {
          return faker.random.float()
        },
        capUnit() {
          return faker.random.number()
        }
      })
    },

    seeds(server) {
      server.createList('pagedGovernanceProposal', 60)
      server.create('resourceCandidate')
    },

    routes() {
      this.get('/governance-actions', (schema) => {
        const content = schema.pagedGovernanceProposals.all()
        const items = content.models.map(({ attrs }) => ({ ...attrs }))
        return {
          pagedGovernanceProposals: {
            totalPages: 0,
            totalElements: items.length || 0,
            size: items.length || 0,
            content: items,
            number: items.length,
            sort: {
              sorted: true,
              unsorted: true,
              empty: false
            },
            pageable: {
              page: 0,
              size: 5,
              sort: ['string']
            },
            first: true,
            last: true,
            numberOfElements: 5,
            empty: !items
          }
        }
      })

      this.post('/governance-actions', (schema, request) => {
        const attrs = JSON.parse(request.requestBody)
        console.log(attrs)

        return schema.pagedGovernanceProposals.create(attrs)
      })

      this.get('/tmf-api/resourceCatalogManagement/v2/resourceCandidate', (schema, serialize) => {
        const resourceCandidates = schema.resourceCandidates.all()
        return resourceCandidates
      })

      this.get('/tmf-api/resourceCatalogManagement/v2/resourceCandidate/:id', (schema, request) => {
        const id = request.params.id
        return [
          {
            '@baseType': 'string',
            '@schemaLocation': 'string',
            '@type': 'string',
            category: [
              {
                '@baseType': 'string',
                '@schemaLocation': 'string',
                '@type': 'string',
                href: 'string',
                id: 'string',
                name: 'string',
                version: 'string'
              }
            ],
            description: 'ubiTest new resource candidate',
            href: 'http://localhost:8080/tmf-api/resourceCatalogManagement/v2/resourceCandidate/a0af6efd-840a-438c-b9f0-f449ba2039d4',
            id: 'a0af6efd-840a-438c-b9f0-f449ba2039d4',
            lastUpdate: '2021-03-17T16:17:36.918Z',
            lifecycleStatus: 'pending',
            name: 'ubiTest',
            resourceSpecification: {
              '@baseType': 'string',
              '@referredType': 'string',
              '@schemaLocation': 'string',
              '@type': 'string',
              href: 'string',
              id: 'string',
              name: 'string',
              version: 'string'
            },
            validFor: {
              endDateTime: '2022-03-17T16:17:36.918Z',
              startDateTime: '2021-03-17T16:17:36.918Z'
            },
            version: '0.1.0'
          }
        ]
      })

      /*  this.get('/resources', (schema, request) => {
        const attrs = JSON.parse(request.requestBody)
        console.log(attrs)
        return schema.resources.all()
      })

      this.get('/resources/:id', (schema, request) => {
        const id = request.params.id
        console.log('id')
        return schema.resources.find(id)
        // new Response(400, { some: 'header' }, { errors: [ 'name cannot be blank'] });
      })

      this.post('/resources', (schema, request) => {
        const attrs = JSON.parse(request.requestBody)
        console.log(attrs)
        return schema.resources.create(attrs)
        // new Response(400, { some: 'header' }, { errors: [ 'name cannot be blank'] });
      })

      this.post('/resources/:id/category', (schema, request) => {
        const attrs = JSON.parse(request.requestBody)
        console.log(attrs)
        return schema.category.create(attrs)
        // new Response(400, { some: 'header' }, { errors: [ 'name cannot be blank'] });
      }) */
    }
  })

  return server
}
