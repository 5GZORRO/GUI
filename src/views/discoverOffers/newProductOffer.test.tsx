// @ts-nocheck
import React from 'react'
import TestRenderer from 'react-test-renderer'
import NewProductOffer from './newProductOffer'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import {
  QueryClient,
  QueryClientProvider
} from 'react-query'
import { makeServer } from 'server'

const history = createMemoryHistory()

describe('New Product Offer', function () {
  const queryClient = new QueryClient()
  let server

  beforeEach(() => {
    server = makeServer({ environment: 'test' })
  })

  afterEach(() => {
    server.shutdown()
  })
  it('Snapshot', () => {
    const tree = TestRenderer.create(
      <Router history={history}>
        <QueryClientProvider client={queryClient}>
          <NewProductOffer />
        </QueryClientProvider>
      </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
