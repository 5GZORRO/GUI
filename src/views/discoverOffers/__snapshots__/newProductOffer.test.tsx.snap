// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Product Offer Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="mb-5 row"
  >
    <div
      className="col"
    >
      <h2>
        New Product Offer
      </h2>
    </div>
  </div>
  <div
    className="card"
  >
    <header
      className="card-header"
    >
      <h5>
        Resource Candidate
      </h5>
    </header>
    <div
      className="card-body"
    >
      <div>
        <div
          className="row my-2 mx-0"
        >
          <div
            className="col-sm-6 form-inline p-0 c-datatable-filter"
          >
            <label
              className="mfe-2"
            >
              Filter:
            </label>
            <input
              aria-label="table filter input"
              className="form-control"
              onChange={[Function]}
              onInput={[Function]}
              placeholder="type string..."
              type="text"
              value=""
            />
            <svg
              className="c-icon mfs-2 transparent"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<polygon fill='var(--ci-primary-color, currentColor)' points='40 16 40 53.828 109.024 136 150.815 136 76.896 48 459.51 48 304 242.388 304 401.373 241.373 464 240 464 240 368 208 368 208 496 254.627 496 336 414.627 336 253.612 496 53.612 496 16 40 16' class='ci-primary'/><polygon fill='var(--ci-primary-color, currentColor)' points='166.403 248.225 226.864 187.763 204.237 165.135 143.775 225.597 83.313 165.135 60.687 187.763 121.148 248.225 60.687 308.687 83.313 331.314 143.775 270.852 204.237 331.314 226.864 308.687 166.403 248.225' class='ci-primary'/>",
                }
              }
              onClick={[Function]}
              onKeyUp={[Function]}
              role="img"
              tabIndex={null}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </div>
      </div>
      <div
        className="position-relative table-responsive"
      >
        <table
          className="table table-hover"
        >
          <thead>
            <tr>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  
                </div>
              </th>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  Name
                </div>
              </th>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  Type
                </div>
              </th>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  Category
                </div>
              </th>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  Resource Specification
                </div>
              </th>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  Valid For
                </div>
              </th>
              <th
                className="position-relative pr-4"
                onClick={[Function]}
                style={
                  Object {
                    "overflow": "hidden",
                    "verticalAlign": "middle",
                    "width": "1%",
                  }
                }
              >
                <div
                  className="d-inline"
                >
                  
                </div>
              </th>
            </tr>
            <tr
              className="table-sm"
            >
              <th
                className=""
              />
              <th
                className=""
              >
                <input
                  aria-label="column name: 'name' filter input"
                  className="form-control form-control-sm"
                  onChange={[Function]}
                  onInput={[Function]}
                  value=""
                />
              </th>
              <th
                className=""
              >
                <input
                  aria-label="column name: 'categoryType' filter input"
                  className="form-control form-control-sm"
                  onChange={[Function]}
                  onInput={[Function]}
                  value=""
                />
              </th>
              <th
                className=""
              >
                <input
                  aria-label="column name: 'categoryName' filter input"
                  className="form-control form-control-sm"
                  onChange={[Function]}
                  onInput={[Function]}
                  value=""
                />
              </th>
              <th
                className=""
              >
                <input
                  aria-label="column name: 'resourceSpecificationName' filter input"
                  className="form-control form-control-sm"
                  onChange={[Function]}
                  onInput={[Function]}
                  value=""
                />
              </th>
              <th
                className=""
              >
                <input
                  aria-label="column name: 'validFor' filter input"
                  className="form-control form-control-sm"
                  onChange={[Function]}
                  onInput={[Function]}
                  value=""
                />
              </th>
              <th
                className=""
              />
            </tr>
          </thead>
          <tbody
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <tr>
              <td
                colSpan={7}
              >
                <div
                  className="text-center my-5"
                >
                  <h2>
                    No items 
                    <svg
                      className="c-icon c-icon-custom-size text-danger mb-2"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path fill='var(--ci-primary-color, currentColor)' d='M425.706,86.294A240,240,0,0,0,86.294,425.705,240,240,0,0,0,425.706,86.294ZM256,48A207.1,207.1,0,0,1,391.528,98.345L98.345,391.528A207.1,207.1,0,0,1,48,256C48,141.309,141.309,48,256,48Zm0,416a207.084,207.084,0,0,1-134.986-49.887l293.1-293.1A207.084,207.084,0,0,1,464,256C464,370.691,370.691,464,256,464Z' class='ci-primary'/>",
                        }
                      }
                      role="img"
                      viewBox="0 0 512 512"
                      width="30"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </h2>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className=""
          style={
            Object {
              "backgroundColor": "rgb(255,255,255,0.4)",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <div
            style={
              Object {
                "left": "50%",
                "position": "absolute",
                "top": "50%",
                "transform": "translateX(-50%) translateY(-50%)",
              }
            }
          >
            <div
              aria-hidden="false"
              aria-label="Loading"
              className="spinner-grow spinner-grow-lg text-primary"
              role="status"
            />
          </div>
        </div>
      </div>
      <nav
        aria-label="pagination"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <ul
          className="pagination justify-content-start"
        >
          <li
            className="page-item disabled"
          >
            <a
              aria-disabled={true}
              aria-label="Go to first page"
              className="disabled page-link"
              href="#"
              onClick={[Function]}
              rel={null}
            >
              «
            </a>
          </li>
          <li
            className="page-item disabled"
          >
            <a
              aria-disabled={true}
              aria-label="Go to previous page"
              className="disabled page-link"
              href="#"
              onClick={[Function]}
              rel={null}
            >
              ‹
            </a>
          </li>
          <li
            className="active page-item"
          >
            <a
              aria-label="Current page 1"
              className="page-link"
              href="#"
              onClick={[Function]}
              rel={null}
            >
              1
            </a>
          </li>
          <li
            className="page-item disabled"
          >
            <a
              aria-disabled={true}
              aria-label="Go to next page"
              className="disabled page-link"
              href="#"
              onClick={[Function]}
              rel={null}
            >
              ›
            </a>
          </li>
          <li
            className="page-item disabled"
          >
            <a
              aria-disabled={true}
              aria-label="Go to last page"
              className="disabled page-link"
              href="#"
              onClick={[Function]}
              rel={null}
            >
              »
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    className="d-flex flex-row-reverse mb-5"
  >
    <button
      className="text-uppercase px-5 btn btn-gradient disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      next
    </button>
    <button
      className="text-uppercase px-5 mr-3 btn btn-outline-white"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
